src/test/groovy/de/dkfz/roddy/execution/jobs/cluster/pbs/PBSJobManagerSpec.groovy

```diff
@@ -6,326 +6,245 @@
-'''
-
-    void setup() {
-        JobManagerOptions parms = JobManagerOptions.create().build()
-        TestExecutionService testExecutionService = new TestExecutionService("test", "test")
-        manager = new PBSJobManager(testExecutionService, parms)
-    }
-
-    void "test processQstatOutputFromXML with queued job"() {
+class PBSJobManagerSpec extends JobManagerImplementationBaseSpec<PBSJobManager> {
+
+    /**
+     * convertResourceSet is implemented in ClusterJobManager but calls methods in the actual JobManager instance
+     */
+    void "test convertResourceSet"(String expected, Integer mem, Integer cores, Integer nodes, String walltime) {
+        given:
+        BufferValue _mem = mem ? new BufferValue(mem, BufferUnit.m) : null
+        TimeUnit _walltime = walltime ? new TimeUnit(walltime) : null
+        ResourceSet test = new ResourceSet(ResourceSetSize.l, _mem, cores, nodes, (TimeUnit) _walltime, null, null, null)
+
+        expect:
+        createJobManager().convertResourceSet(null, test).processingCommandString == expected
+
+        where:
+        expected                                                | mem  | cores | nodes | walltime
+        "-l mem=1024M -l walltime=00:01:00:00 -l nodes=1:ppn=2" | 1024 | 2     | 1     | "h"
+        "-l walltime=00:01:00:00 -l nodes=1:ppn=1"              | null | null  | 1     | "h"
+        "-l mem=1024M -l nodes=1:ppn=2"                         | 1024 | 2     | null  | null
```

Comment:
nodes == null? nul means use the default, right?

For clarity I'd prefer to see the inputs on the left and the expect on the right (but that is nitpicking :-P)